test/functional/test_framework/script.py

```diff
@@ -731,3 +731,12 @@ def test_bn2vch(self):
         self.assertEqual(bn2vch(0xFFFFFFFF), bytes([0xFF, 0xFF, 0xFF, 0xFF, 0x00]))
         self.assertEqual(bn2vch(123456789), bytes([0x15, 0xCD, 0x5B, 0x07]))
         self.assertEqual(bn2vch(-54321), bytes([0x31, 0xD4, 0x80]))
+    def test_cscriptnum_encoding(self):
+        # round-trip negative and multi-byte CScriptNums to catch python regression
+        self.assertEqual(CScriptNum.decode(CScriptNum.encode(CScriptNum(1500))), 1500)
+        self.assertEqual(CScriptNum.decode(CScriptNum.encode(CScriptNum(-1500))), -1500)
+        self.assertEqual(CScriptNum.decode(CScriptNum.encode(CScriptNum(-1))), -1)
+    def test_cscriptnum_encoding_large(self):
```

Comment:
Question: I'm not too familiar with what "catch python regression" means 😕  so I'm kind of confused about the difference between`test_cscriptnum_encoding` and `test_cscriptnum_encoding_large`. Could they be combined in the `values` array?

Reply:
"python regression" is an odd way to say "regression in the scriptnum encoding (which is written in python and not c++)"

Reply:
So you are probably right and they can be combined in one list.

Reply:
I think you can remove " to catch python regression" from the comment